components/resources/quizzes/shared/SkillSelections.tsx

```diff
@@ -37,22 +71,43 @@ const SkillSelections: React.FC<SkillSelectionsProps> = ({
         </div>
         <div className="text-lg font-semibolds px-3">{body}</div>
         <div className="flex flex-col w-full max-w-4xl mx-auto">
-          {selections.map((selection, index) => (
-            <div
-              key={index}
-              className={`flex items-start justify-start w-full px-4 py-2 my-2 cursor-pointer ${
-                selected.includes(selection)
-                  ? "bg-violet-300 text-black-500 border-2 border-black-500 rounded-xl"
-                  : "bg-white text-black-600 border-2 border-black-500 rounded-xl"
-              }`}
-              onClick={() => handleSelection(selection)}
-            >
-              {selections[index]}
-            </div>
-          ))}
+          {selections.map((selection, index) => {
+            const isSelected = currentSelections.includes(selection);
+            const className = [
+              "flex",
+              "items-start",
+              "justify-start",
+              "w-full",
+              "px-4",
+              "py-2",
+              "my-2",
+              "cursor-pointer",
+              "text-black-600",
+              "border-2",
+              "border-black-500",
+              "rounded-xl",
+              isSelected ? "bg-violet-300" : "bg-white",
+            ].join(" ");
+            return (
+              <div
+                key={index}
+                className={className}
+                onClick={() => handleSelection(selection)}
+              >
+                {selection}
+              </div>
+            );
+          })}
         </div>
         <div className="mt-4">
-          <Button label="Next" onClick={onNextClick} backgroundColor="yellow" />
+          <Button
+            label="Next"
+            onClick={() => {
+              handleStage();
```

Comment:
You can't use shorthand syntax here when calling two functions so this is correct